modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCacheMapEntry.java

```diff
@@ -5763,6 +5766,11 @@ private CacheDataRow checkRowExpired(CacheDataRow row) throws IgniteCheckedExcep
      *
      */
     private static class AtomicCacheUpdateClosure implements IgniteCacheOffheapManager.OffheapInvokeClosure {
+
+        /** Flag for disabling interceptor on updates/deletes by DR events. */
+        private final boolean disableInterceptorOnDrEvts =
```

Comment:
You don't need another one boolean field in each atomic update closure.

Better move the property to GridCacheDrManager: `org.apache.ignite.internal.processors.cache.dr.GridCacheDrManager#cacheInterceptorDisabled`